src/ui/EditorNS/jsproxy.cpp

```diff
@@ -39,6 +39,10 @@ void JsToCppProxy::sendEditorEvent(QString msg, QVariant data)
     if (msg == "J_EVT_READY" && m_ready == false) {
         m_ready = true;
         pushQueuedMessages();
+    }else if(msg == "J_EVT_CLEAN_CHANGED") {
+        m_values["clean"] = data;
+    }else if(msg == "J_EVT_CURRENT_LANGUAGE_CHANGED") {
+        m_values["language"] = data;
```

Comment:
J_EVT_CLEAN_CHANGED and J_EVT_CURRENT_LANGUAGE_CHANGED don't belong here because we don't want to know about all possible events here. J_EVT_READY was only a special case.

There's still room for improvements but I actually prefer how it was done before... you can revert this commit (fcd8dd022729615d94f7d81eb22748e9f15b19e6)